vault/logical_system_helpers.go

```diff
@@ -7,61 +7,31 @@ import (
 )
 
 // tuneMount is used to set config on a mount point
-func (b *SystemBackend) tuneMountTTLs(path string, me *MountEntry, newDefault, newMax *time.Duration) error {
-	meConfig := &me.Config
+func (b *SystemBackend) tuneMountTTLs(path string, me *MountEntry, newDefault, newMax time.Duration) error {
+	zero := time.Duration(0)
 
-	if newDefault == nil && newMax == nil {
-		return nil
-	}
-	if newDefault == nil && newMax != nil &&
-		*newMax == meConfig.MaxLeaseTTL {
-		return nil
-	}
-	if newMax == nil && newDefault != nil &&
-		*newDefault == meConfig.DefaultLeaseTTL {
-		return nil
-	}
-	if newMax != nil && newDefault != nil &&
-		*newDefault == meConfig.DefaultLeaseTTL &&
-		*newMax == meConfig.MaxLeaseTTL {
-		return nil
-	}
+	switch {
+	case newDefault == zero && newMax == zero:
+		// No checks needed
 
-	if newMax != nil && newDefault != nil && *newMax < *newDefault {
-		return fmt.Errorf("new backend max lease TTL of %d less than new backend default lease TTL of %d",
-			int(newMax.Seconds()), int(newDefault.Seconds()))
-	}
+	case newDefault == zero && newMax != zero:
+		// No default/max conflict, no checks needed
 
-	if newMax != nil && newDefault == nil {
-		if meConfig.DefaultLeaseTTL != 0 && *newMax < meConfig.DefaultLeaseTTL {
-			return fmt.Errorf("new backend max lease TTL of %d less than backend default lease TTL of %d",
-				int(newMax.Seconds()), int(meConfig.DefaultLeaseTTL.Seconds()))
-		}
-	}
+	case newDefault != zero && newMax == zero:
+		// No default/max conflict, no checks needed
 
-	if newDefault != nil {
-		if meConfig.MaxLeaseTTL == 0 {
-			if newMax == nil && *newDefault > b.Core.maxLeaseTTL {
-				return fmt.Errorf("new backend default lease TTL of %d greater than system max lease TTL of %d",
-					int(newDefault.Seconds()), int(b.Core.maxLeaseTTL.Seconds()))
-			}
-		} else {
-			if newMax == nil && *newDefault > meConfig.MaxLeaseTTL {
-				return fmt.Errorf("new backend default lease TTL of %d greater than backend max lease TTL of %d",
-					int(newDefault.Seconds()), int(meConfig.MaxLeaseTTL.Seconds()))
-			}
+	case newDefault != zero && newMax != zero:
```

Comment:
This switch statement might be cleaner if it only had this case and a default case. Unless you like the verbosity of the other statements + comments.

Reply:
In this instance I found it useful to think through the different cases and leave appropriate comments.